src/components/views/admin/BuilderView.tsx

```diff
@@ -15,33 +18,25 @@ import { useAppDataContext } from '../../context/AppDataContext';
 
 const BuilderView: FC = () => {
   const { appDataArray } = useAppDataContext();
-  const [groupedObjects, setGroupedObjects] = useState<
-    { memberId: string; objects: any[] }[]
-  >([]);
-  const [selectedType, setSelectedType] = useState('');
-  const groupObjectsByMemberId = (): void => {
-    const groupedData: { [key: string]: any[] } = {};
-    appDataArray.forEach((data) => {
-      const { memberId } = data;
-      if (memberId in groupedData) {
-        groupedData[memberId].push(data);
-      } else {
-        groupedData[memberId] = [data];
-      }
-    });
+  const groupObjectsByMemberId = (): List<{
+    memberId: string;
+    objects: any[];
+  }> => {
```

Comment:
Is the shape returned by `groupBy` not convenient for you to use ? 
It is a `Map` (similar to an object) where you have the keys as `memberId` and the values are a list of all appData related to that `memberId`.